README.md
## Jutten 介紹

![Juuten_Icon](public/icon/Juuten_Icon_128.png)

[Demo](https://drive.google.com/file/d/1zYdZe_0dbDFHZ9wPQDD2CAmf9IV_dOHN/view?usp=share_link)

如果你需要一個可以隨時快速筆記，快速把中意的句子或文章記錄起來，Juuten 可以滿足你的需求

### 未來預計新增
- 黑暗模式
- 更改字體大小
- 更改字體顏色

### 隨時記錄

隨時把所需的文字加入至 Juuten

![gif1](public/gif/1.gif)

### 隨時筆記

可以隨時記錄你所想的東西

![gif2](public/gif/2.gif)

### 自由的創建編輯和排序

照你所想自由地進行分類和排序

![gif3](public/gif/3.gif)

## 使用技術

- `React` 前端框架
- `Redux` 狀態管理
- `React Router` 路由工具
- `styled-components` css-in-js
- `Redux-Thunk` 處理非同步狀態
- `Draftjs` 筆記文本編輯器
- `React-beautiful-dnd` dnd(拖移)組件
- `webpack` 打包整合工具

## 代碼簡介

> 以下代碼皆為精簡後的版本

### Draft.js

使用 Draft.js 創立組件後利用 `forwardRef` 配合 `useImperativeHandle` 將函式和 State 向前傳送，讓父組件也能使用，之後要新增函數或進行改動也相當容易

```jsx
/* DraftComponent.jsx */
const DraftComponent = memo(
  forwardRef(
    (
      {
        item = "",
        readOnly = false,
        setInlineStyle,
        open = false,
        autoSave = {},
      },
      ref
    ) => {
      const mainRef = useRef(null);
      const [editorState, setEditorState] = useState(() => {
        if (item)
          return EditorState.createWithContent(
            convertFromRaw(JSON.parse(item))
          );
        else return EditorState.createEmpty();
      });

      function onChange(state) {
        setEditorState(state);
      }

      useImperativeHandle(
        ref,
        () => ({
          editorState: editorState,
          toggleStyle: (style, e) => {
            e.preventDefault();
            const newState = RichUtils.toggleInlineStyle(editorState, style);
            nowInlineStyle(newState);
            onChange(newState);
          },
          autoFocus: () => {
            onChange(EditorState.moveFocusToEnd(editorState));
          },
        }),
        [editorState]
      );
      return (
        <Editor editorState={editorState} onChange={onChange} ref={mainRef} />
      );
    }
  )
);
```

### React Thunk

使用`React Thunk` 利用 Goole API `Chrome storage local` 獲取資料，之後使用回調進行畫面跳轉

```javascript
/* FolderBlock.jsx */
function goIntoFolder(e, item) {
  e.stopPropagation();
  const fn = () => navigate(`/collection/${item.key}`);
  dispatch(
    addFetchData({
      item,
      fn,
    })
  );
}

/* collectionSlice.js */
const thunkData = createAsyncThunk(
  "folder/fetchFolderData",
  async ({ item, fn }) => {
    const value = await fetchData(item.key, []);
    return { item, value, fn };
  }
);
export const CollectionSlice = createSlice({
  name: "collection",
  initialState: {
    /* ... */
  },
  reducers: {
    /* ... */
  },
  extraReducers: {
    [thunkData.fulfilled]: (state, action) => {
      const { item, value, fn = () => {} } = action.payload;
      setDataToLocal("Juuten_Navigate_History", item);
      fn();
      return {
        ...state,
        folderData: item,
        folderId: item.key,
        isHistoryLoad: true,
        [item.key]: value,
      };
    },
  },
});
```

### Search Note

在所有的筆記內查找是否有符合輸入的文字，如果有把所有相同的文字打上 Highlight

```javascript
/* FoundData.jsx */
function findData() {
  const target = inputRef.current?.value.toUpperCase();
  if (!target) {
    setFoundData([]);
    return;
  }
  const newData = data
    .map((item) => {
      // 先篩選符合條件的註解(comment)，並加上 highlight 的樣式
      const comment = item.comment
        .filter((item2) => isMsgHas(item2.msg, target))
        .map((comm) => ({ ...comm, msg: highlightMsg(comm.msg, target) }));

      // 如果主要文字也符合搜尋條件，也要加上 highlight 的樣式
      const _data = {
        ...item,
        comment: comment,
        msg: isMsgHas(item.msg, target)
          ? highlightMsg(item.msg, target)
          : comment.length
          ? highlightMsg(item.msg, target)
          : "",
      };
      // 排除沒有搜尋結果的資料
      return !!_data.msg || !!_data.comment.length ? _data : null;
    })
    .filter((item) => item !== null)
    // 將結果依照資料夾分類
    .reduce((outputObject, { folderName, ...rest }) => {
      if (!outputObject[folderName]) outputObject[folderName] = [];
      outputObject[folderName].push({ ...rest });
      return outputObject;
    }, {});
  // 更新搜尋結果
  setFoundData(newData);
}

function isMsgHas(msg, target) {
  /**
   * 檢查 `msg` 是否包含 `target` 字符串
   */
  const contentState = convertFromRaw(JSON.parse(msg));
  const plainText = contentState.getPlainText().toUpperCase();
  return plainText.includes(target);
}

function highlightMsg(msg, target) {
  /**
   * 將給定的消息中的目標文本高亮
   */
  let editorState = EditorState.createWithContent(
    convertFromRaw(JSON.parse(msg))
  );
  const blocks = editorState.getCurrentContent().getBlockMap();
  const regex = new RegExp(target, "g");

  // 每個Block檢測是否有相同文字，如果有打上Highlight
  blocks.forEach((block) => {
    const text = block.getText().toUpperCase();
    let matchArr = regex.exec(text);
    while (matchArr !== null) {
      editorState = RichUtils.toggleInlineStyle(
        EditorState.forceSelection(
          editorState,
          editorState.getSelection().merge({
            anchorKey: block.getKey(),
            focusKey: block.getKey(),
            anchorOffset: matchArr.index,
            focusOffset: matchArr.index + target.length,
          })
        ),
        "HIGHLIGHT"
      );
      matchArr = regex.exec(text);
    }
  });
  return JSON.stringify(convertToRaw(editorState.getCurrentContent()));
}
```

## 聲明

此軟體完全由 Zeke 進行開發 如有 Bug 歡迎通知我

Github: [Zeke](https://github.com/JikeLuo)

E-Mail: traveller87072@gmail.com

Version: 0.22.3



CHANGELOG.md
## Juuten v0.22.3 - 2023/5/1
- debug .htm => .html

## Juutn v0.22.2 - 2023/5/1
- OPT 路徑alias
- OPT barPage說明更新

## Juuten v0.22.1 - 2023/4/30

- Debug 創建新筆記後依然會讀取舊 autoSave 而不清乾淨
- Debug 在 Juuten 會快捷鍵的問題

## Juuten v0.22.0 - 2023/4/29

- 新增 鍵盤的快捷鍵 redo undo bold italic...
- Debug navbar 無法收起的問題

## Juuten v0.22.0 - 2023/4/29
- 新增 鍵盤的快捷鍵 redo undo bold italic...
- Debug navbar無法收起的問題

## Juuten v0.21.0 - 2023/4/28

- 新增 紀錄上次所在資料夾
- 新增 按下 TAB 可推進 4 格空白
- debug 一些 draftjs
- 優化 使用 hook 優化代碼

## Juuten v0.20.0 - 2023/4/27

- 新增 Bug 回報頁面
- DEBUG 修復 10 多項的 BUG
- 優化 使用者體驗

## Juuten v0.19.0 - 2023/4/23

- 新增 select 小工具
- 新增 icon
- 優化 all reducer to pure reducer
- 優化 search function

## Juuten v0.18.0 - 2023/04/20

- Debug 寬度縮小主色調 box 跑位
- Debug Navbar 顯示 undefined
- 新增 編輯過程自動存檔
- 新增 刪除 note 時的二度確認
- 修改 UI 樣式小改，Texting Toolbar UX 修改
- 優化 使用 memo 優化大部分組件，避免過度 re-render

## Juuten v0.17.0 - 2023/04/16

- 新增 搜尋筆記
- 新增 設置 Juuten 外觀
- 新增 匯出成 TXT HTMLs
- 優化 小組件 進度 40%

## Juuten v0.16.0 - 2023/04/11

- 可拖移區域全完成
- 優化 folder 版面的配置

## Juuten v0.15.0 - 2023/04/09

- 新增 React-beautiful-dnd 讓筆記可以自由地移動
- 優化 讓字體或 ICON 大小定義在一組變數中

## Juuten v0.14.0 - 2023/04/06

- 新增 Draft.js package

## Juuten v0.13.0 - 2023/03/30

- 優化 collection 獨立各 reducer 功能

## Juuten v0.12.0 - 2023/03/29

- 修改 工具列位置變換
- Debug 無法新增註記的問題

## Juuten v0.11.0 - 2023/03/28

- 修改 文字區工具列
- 修改 文章編輯後的儲存方式
- 修改 暫存區和錦集的移動

## Juuten v0.10.0 - 2023/03/27

- 新增 資料夾編輯的工具列

## Juuten v0.9.0 - 2022/09/24

- 移除 留言的嵌套功能
- 修改 儲存的方式
- 修改 暫存區匯入匯出的 Icon

## Juuten v0.8.0 -2022/09/23

- 新增 暫存區收藏加入資料夾
- 新增 資料夾筆記加入暫存區
- 優化 編輯時 autoFocus
- 優化 編輯時高度跟隨內文高度

## Jutten v0.7.0 -2022/09/22

- 新增 新增評論功能
- 新增 刪除筆記、評論功能
- 新增 修改筆記、評論功能
- 修改 Note 資料結構
- 修改 日期顯示位置

## Juuten v0.6.0 -2022/09/21

- 新增 新增資料夾動畫
- 新增 刪除資料夾功能
- 新增 加入新筆記
- 新增 加入新筆記的動畫
- 修改 資料夾檔案結構
- 優化 進入資料夾的邏輯
- 優化 非同步獲取 chrome Storage 資料
- 優化 新增資料夾動畫

## Juuten v0.5.0 -2022/09/17

- 新增 資料夾檔案資料儲存到 storage sync

## Juuten v0.4.0 -2022/09/17

- 新增 新增資料夾
- 新增 更改資料夾顏色
- 新增 字體顏色隨著資料夾顏色改變
- 新增 Toolbar 出現動畫
- 新增 判別資料夾名稱重複
- 新增 犯別違法資料夾名稱
- 新增 資料夾外觀
- 修改 redux 新的更改 state 方式
- 修改 字體引入方式
- 修改 資料夾結構
- 修改 file-loader to asset/resource
- 修改 資料夾的資料結構

## Juuten v0.3.0 -2022/09/16

- 新增 退出前儲存提醒
- 改善 collection 的排版
- 新增 note 修改
- 新增 儲存刪除 icon
- 新增 已到最底
- 新增 編輯中 小動畫
- 修改 storage 存有數量 icon
- 修改 字體
- 新增 創建日期顯示
- 修改 紅色色卡
- 新增 編輯時動畫

## Juuten v0.2.0 -2022/09/15

- 新增 folder 頁面
- 新增 collection 頁面
- 新增 collection navbar icon 和 storage 開關動作
- 新增 storage 儲存數量
- 搭建 collection 筆記的樣式


src/assets/fakeData.js
const Juuten_folderLists = [
    {
        key: '908084',
        name: '20221015_中華料理',
        folderColor: 'pink',
        font: false,
        createDate: '',
        tags: []
    },
    {
        key: '959218918',
        name: '20090801_台北旅行',
        folderColor: '#1f1e33',
        font: false,
        createDate: '',
        tags: []
    },
    {
        key: '445954118',
        name: '19980720_我的出生aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
        folderColor: '#1f1e33',
        font: false,
        createDate: '',
        tags: []
    },
    {
        key: '089aas4asas890',
        name: '20210519_日本料理',
        folderColor: 'skyblue',
        font: false,
        tags: []
    },
    {
        key: '0sss890',
        name: '關於如何使用Typescript達到效能最佳化',
        folderColor: '#1f1e33',
        font: false,
        tags: []
    },
    {
        key: 'N1',
        name: '關於如何使用Typescript達到效能最佳化 aaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
        folderColor: '#1f1e33',
        font: false,
        tags: []
    },
    {
        key: 'sus',
        name: 'sas',
        folderColor: 'yellow',
        font: false,
        tags: []
    },
    {
        key: '0894sdeg890',
        name: 'sas',
        folderColor: 'yellow',
        font: false,
        tags: []
    },
    {
        key: '089aa4eg89sw0',
        name: 'sas',
        folderColor: 'yellow',
        font: false,
        tags: []
    },
]

const folder = [
    {
        "key": "Juuten_1682663829855",
        "name": "+ 可創建新的彙編",
        "folderColor": "#ffd14d",
        "font": true,
        "createDate": "2023/3/28 14:37",
        "tags": []
    },
    {
        "key": "Juuten_1682663857766",
        "name": "點選右側可開啟彙編工具列 ->",
        "folderColor": "#ffd14d",
        "font": true,
        "createDate": "2023/3/28 14:37",
        "tags": []
    },
    {
        "key": "Juuten_1682663775499",
        "name": "<- 抓移左側可自由排序彙編",
        "folderColor": "#ffd14d",
        "font": true,
        "createDate": "2023/3/28 14:36",
        "tags": []
    },
    {
        "key": "Juuten_1682663998192",
        "name": "筆記的介紹及使用方法（雙擊我進入）",
        "folderColor": "#62dbc8",
        "font": true,
        "createDate": "2023/3/28 14:39",
        "tags": []
    },
    {
        "key": "Juuten_1682668691074",
        "name": "注意 : 如果移除該APP，APP內所儲存的資料將一併消失",
        "folderColor": "#ff5757",
        "font": false,
        "createDate": "2023/3/28 15:58",
        "tags": []
    }
]

const data =  [
    '下方工具欄的介紹\n' +
    '1. 搜尋 : 可在所有筆記間查詢\n' +
    '2. 新增 : 可新增筆記，如果在主頁進行新增則會把筆記儲存到 暫存區中\n' +
    '3. 暫存區 : 可放在主頁 "新增的筆記" 和 "加入至 Juuten" 的筆記\n' +
    '4. 設定 : 可進行APP的顏色或字體大小的基本設定',

    '  筆記的基本介紹\n' +
    '1. 新增筆記 : 點選下方 + 可創建一則新筆記\n' +
    '2. 移動筆記 : <- 抓取筆記左側可進行拖移\n' +
    '3. 編輯筆記 : 點選右上 "鉛筆的 ICON" 可編輯該筆記\n' +
    '4. 新增註記 : 點選右上 "留言的 ICON" 可對每則筆記新增無數個註記\n' +
    '2023/3/28 15:37\n' +
    '1. 基本功能 : 每個註記的基本功能如同一般的筆記\n' +
    '2. 移動註記 : 點選該 "筆記的編輯按鈕" 後，註記左側會升起可抓移的區域，之後可自由排序註記的位置\n' +
    '\n' +
    '注意: 不同筆記間的註記無法自由移動',

    '右上的工具列介紹\n' +
    '1. 註記 : 可以創建一則新註記\n' +
    '2. 編輯 : 可以編輯此筆記的文本\n' +
    '3. 下推 : 可以把筆記推進至 暫存區 (下方工具欄右2) 讓筆記可在彙編間自由移動',

    '在網頁新增重點筆記\n' +
    '1. 在網頁（少數無法使用）選取文字\n' +
    '2. 點選隨之出現的 Juuten Icon 或是 右鍵選擇 "加入至 Juuten" 按鈕\n' +
    '3. 儲存的筆記會存至 Juuten 的暫存區\n' +
    '2023/3/28 15:50\n' +
    '暫存區介紹\n' +
    '1. 基本功能 : 和筆記一樣可進行編輯和位置移動，但不能新增註記\n' +
    '2. 上推 : 點選右上的 ICON 即可把該則筆記上推至目前所在的彙編內\n' +
    '3. 暫存 : 可儲存 "加入至Juuten" 所新增的筆記',
]

const N1 = [
    {
        key: 'Juuten_999556',
        type: 'collection',
        favIconUrl: 'https://img.icons8.com/material/344/globe--v2.png',
        url: 'https://www.youtube.com/watch?v=HSW6qfRUT9w',
        currentDate: '2022/10/10 12:11',
        msg: `{"blocks":[{"key":"an8qb","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}`,
        isOpenComment: false,
        comment: [
            {
                key: 'Juuten_255',
                type: 'comment',
                currentDate: '2022/10/10 12:11',
                msg: '{"blocks":[{"key":"70fru","text":"最慘的難，去了已經看不到俄羅斯的未來在哪裡...","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":44,"length":3,"style":"CODE"},{"offset":106,"length":3,"style":"CODE"}],"entityRanges":[],"data":{}}],"entityMap":{}}',
                comment: []
            }, {
                key: 'Juuten_2s55',
                type: 'comment',
                currentDate: '2022/10/10 12:11',
                msg: '{"blocks":[{"key":"a1o9s","text":"xxx jsisja klslsp;a xxx okokoko xx kook9jxx j9j9kpxkpkopkxxkopkopkmxxxipjpkoopxxkpokopx,","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
                comment: []
            },
            {
                key: 'Juuten_344',
                type: 'comment',
                currentDate: '2022/10/10 12:11',
                msg: '{"blocks":[{"key":"1ebtn","text":"對絕大部分整個烏克嚴重悟（或許吧）","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
                comment: []
            },
        ]
    },
    {
        key: 'Juuten_1556556',
        type: 'collection',
        favIconUrl: 'https://img.icons8.com/material/344/globe--v2.png',
        url: 'https://www.youtube.com/watch?v=HSW6qfRUT9w',
        currentDate: '2022/10/10 12:11',
        msg: `{"blocks":[{"key":"8s8rf","text":"本影片的資料主要來源為網上與社群媒體，資料眾","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"fuqhl","text":"多未能一一查","type":"header-three","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"ctr4","text":"，資訊並不一定百分之百正確，內容僅供參考。志在分享，如有誤點，歡迎指正。","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}`,
        isOpenComment: true,
        comment: [
            {
                key: 'Juuten_2',
                type: 'comment',
                currentDate: '2022/10/10 12:11',
                msg: '{"blocks":[{"key":"70fru","text":"最慘的是永遠是俄羅斯自己蘇聯時期跑掉一堆人才，導致解體後除了能源軍工外各個產業發展困難，GDP窮到只跟伊朗差不多結果開戰後變本加厲，不只跑掉更多人才，連唯二賺錢的軍火和能源都賣不出去了已經看不到俄羅斯的未來在哪裡...","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":44,"length":3,"style":"CODE"},{"offset":106,"length":3,"style":"CODE"}],"entityRanges":[],"data":{}}],"entityMap":{}}',

            },
            {
                key: 'Juuten_3',
                type: 'comment',
                currentDate: '2022/10/10 12:11',
                msg: '{"blocks":[{"key":"1ebtn","text":"對絕大部分平民來說，哪個地方是不是自己的，與個人利益幾乎完全沒有任何關係，就假如俄羅斯真的奪下整個烏克蘭，中國奪下整個台灣，國內的人民的生活絲毫不會有任何改善，甚至可能承受種種戰爭下的苦果，但對於那些已被嚴重洗腦的人來說講再多也是白費唇舌，只有真正體會了才能幡然醒悟（或許吧）","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',

            },
            {
                key: 'Juuten_5',
                type: 'comment',
                currentDate: '2022/10/10 12:11',
                msg: '{"blocks":[{"key":"3spv9","text":"舉例有些遙\遠\，個人是?看長期國際貿易和貿?易聯盟零關稅加深供應|鏈過度集中，供應堅\韌性逐漸下降一堆國家工業製造都出現斷層，各類產品可能主要前3產國就可以占到市占7~9成，如同投資放在同個籃利潤大風險也大，現在風險發生利潤是要還回去的可能還要倒還，就如同長期政客為了當選低利率，風險發生通膨要全民一起還，俄國出口主要的自然資源又是附近國家很缺的+加上國貿較晚起步還不夠依賴，影響自然會比較少；現在中台貿易從現政府上任一來一年3.5兆漲到7兆台幣都不知道到底是不是假站美國邊，然後台幣升就阻時不時升跌就放跌...真的超獨厚出口資方老闆；供應堅韌性逐漸下降EX:疫情如果在晚個5年爆發搞不好那時台灣國內就已經沒有人生產口罩了","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":67,"length":1,"style":"CODE"},{"offset":77,"length":3,"style":"CODE"},{"offset":168,"length":1,"style":"CODE"},{"offset":208,"length":3,"style":"CODE"},{"offset":214,"length":1,"style":"CODE"},{"offset":248,"length":3,"style":"CODE"},{"offset":272,"length":3,"style":"CODE"},{"offset":282,"length":1,"style":"CODE"}],"entityRanges":[],"data":{}}],"entityMap":{}}',

            },
            {
                key: 'Juuten_599',
                type: 'comment',
                currentDate: '2022/10/10 12:11',
                msg: '{"blocks":[{"key":"2bk71","text":"這篇文?章將/討論如何在 JavaScript 中從數組中刪除未定義的值。該解決方案應返回一個排除未定義值的新數組。","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":1,"length":2,"style":"BOLD"},{"offset":4,"length":2,"style":"BOLD"},{"offset":7,"length":2,"style":"BOLD"},{"offset":11,"length":10,"style":"STRIKETHROUGH"},{"offset":24,"length":10,"style":"ITALIC"},{"offset":39,"length":17,"style":"UNDERLINE"}],"entityRanges":[],"data":{}}],"entityMap":{}}',

            },
        ]
    },
    {
        key: 'Juuten_2asww55',
        type: 'collection',
        currentDate: '2022/10/10 12:11',
        msg: '{"blocks":[{"key":"70fru","text":"最慘的難，去了已經jjjjjjjjj看不到俄羅斯的未來在哪裡...","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":44,"length":3,"style":"CODE"},{"offset":106,"length":3,"style":"CODE"}],"entityRanges":[],"data":{}}],"entityMap":{}}',
        isOpenComment: false,
        comment: []
    }
]


const Juuten_Storage = [
    {
        key: 'Juuten_23',
        msg: '{"blocks":[{"key":"327mv","text":"hi i am collection","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
        type: 'storage',
        url: 'https://www.google.com/search?q=info.selectionText+%27%5Cn%27&sxsrf=APwXEdfs9TbvxEN1H0Zk_lvHvTMKTqtAaA%3A1681923281454&ei=0RxAZLSwG9fmwQOpr77gBA&ved=0ahUKEwj03Ou9tLb-AhVXc3AKHamXD0wQ4dUDCA8&uact=5&oq=info.selectionText+%27%5Cn%27&gs_lcp=Cgxnd3Mtd2l6LXNlcnAQAzIFCAAQogQyBQgAEKIEMgUIABCiBDIFCAAQogQ6CggAEEcQ1gQQsAM6CgghEKABEMMEEAo6BwghEKABEApKBAhBGABQhAdYmwpgxg9oAnABeACAAZIBiAG7ApIBAzIuMZgBAKABAcgBAsABAQ&sclient=gws-wiz-serp',
        favIconUrl: '',
        pageTitle: 'david love lucy',
        currentDate: '2022/10/10 12:10',
        position: 203,
        comment: []
    },
    {
        key: 'Juuten_24',
        msg: '{"blocks":[{"key":"8ot9e","text":"巴哈姆特 GNN 與日本遊戲業界分析公司 MediaCreate 合作，每週定期刊載由 MediaCreate 調查與統計的日本、台灣與韓國地區家用遊樂器主機遊戲軟體的一週銷售排行榜，供玩家參考","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":4,"length":5,"style":"CODE"},{"offset":20,"length":13,"style":"CODE"},{"offset":43,"length":13,"style":"CODE"}],"entityRanges":[],"data":{}}],"entityMap":{}}',
        type: 'storage',
        url: 'https://gnn.gamer.com.tw/detail.php?sn=240417',
        favIconUrl: '',
        pageTitle: 'MediaCreate 10 月 17 日～10 月 23 日一週銷售排行榜 《女神異聞錄 5 皇家版》包辦冠亞季軍',
        currentDate: '2022/10/10 12:10',
        position: 203,
        comment: []
    },
]
export {Juuten_Storage, N1, Juuten_folderLists}